addons/account-switching/userscript.js

```diff
@@ -0,0 +1,111 @@
+const el = (el, properties = {}, children) => {
+  const element = document.createElement(el);
+  Object.keys(properties).forEach((key) => {
+    if (key === "listeners") {
+      Object.keys(properties.listeners).forEach((event) => {
+        element.addEventListener(event, properties.listeners[event]);
+      });
+    } else {
+      element[key] = properties[key];
+    }
+  });
+  if (Array.isArray(children)) {
+    children.forEach((child) => {
+      element.appendChild(child);
+    });
+  } else if (children) {
+    element.appendChild(children);
+  }
+  return element;
+};
+
+export default async function ({ addon, msg, global, console }) {
+  while (true) {
+    const dropdown = await addon.tab.waitForElement("ul.dropdown, ul.user-nav", { markAsSeen: true });
+    await addon.tab.waitForElement(".divider");
+    const username = await addon.auth.fetchUsername();
+
+    function sendBGMsg(type, username) {
+      return async function (event) {
+        if (!event.isTrusted) return;
+        addon.messaging.sendMessage({ type, username }, () => {
+          window.location.reload();
+        });
+      };
```

Comment:
Probably doesn't need async function for now (could the messaging api in this PR be using promises instead of callbacks?)

Reply:
Imo callbacks are better for message apis especially since responses might not come back right away. That's my opinion, if anyone thinks differently, let me know, I can add both callback and promise.

Reply:
yeah, so use promises to get the result inline with `async`/`await`, but if one person wants to use callbacks, they can use `.then`

Reply:
But there's no good reason to use awaits when messages aren't meant to get responce right away.